src/extension/content-script/nostr.js

```diff
@@ -82,12 +83,19 @@ async function init() {
         origin: getOriginData(),
       };
 
+      const account = await api.getAccount();
```

Comment:
why do we load the account here in the message handler, but load it for liquid on the init() function?

Reply:
resolved! extended same code on other providers. keeping account check in listener function helps in executing commands without need of reload

Reply:
Good catch -It should be done in the message handler for each right, so that the user does not need to refresh after the key is set (and also so it checks every time, after a user switches account for example)